src/aead/chacha.rs

```diff
@@ -12,7 +12,7 @@
 // WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION
 // OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
 // CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
-
+#![cfg_attr(target_os = "wasi", allow(unused, dead_code))]
```

Comment:
I'm not understanding why we suddenly have some code becoming dead code, but this doesn't seem right.